cmd/argo-watcher/state/postgres_state.go

```diff
@@ -2,81 +2,90 @@ package state
 
 import (
 	"database/sql"
-	"encoding/json"
 	"fmt"
 	"time"
 
+	"github.com/google/uuid"
+	"github.com/rs/zerolog/log"
+	"gorm.io/datatypes"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+	"gorm.io/gorm/logger"
+
 	"github.com/avast/retry-go/v4"
-	"github.com/golang-migrate/migrate/v4"
 	_ "github.com/lib/pq"
-	"github.com/rs/zerolog/log"
 
-	"github.com/golang-migrate/migrate/v4/database/postgres"
 	_ "github.com/golang-migrate/migrate/v4/source/file"
 
 	"github.com/shini4i/argo-watcher/cmd/argo-watcher/config"
+	"github.com/shini4i/argo-watcher/cmd/argo-watcher/state/state_models"
 	"github.com/shini4i/argo-watcher/internal/models"
 )
 
 type PostgresState struct {
-	db *sql.DB
+	orm *gorm.DB
 }
 
 // Connect establishes a connection to the PostgreSQL database using the provided server configuration.
-// It takes a pointer to a config.ServerConfig parameter and initializes the database connection.
-// The method also performs database migrations using the specified migrations path.
-// If any errors occur during connection establishment or migrations, they are logged and may cause a panic.
-func (state *PostgresState) Connect(serverConfig *config.ServerConfig) {
-	c := fmt.Sprintf("host=%s port=%s user=%s password=%s dbname=%s sslmode=disable", serverConfig.DbHost, serverConfig.DbPort, serverConfig.DbUser, serverConfig.DbPassword, serverConfig.DbName)
+func (state *PostgresState) Connect(serverConfig *config.ServerConfig) error {
+	dsnTemplate := "host=%s port=%s user=%s password=%s dbname=%s sslmode=disable TimeZone=UTC"
+	dsn := fmt.Sprintf(dsnTemplate, serverConfig.DbHost, serverConfig.DbPort, serverConfig.DbUser, serverConfig.DbPassword, serverConfig.DbName)
+
+	// create connection
+	ormConfig := &gorm.Config{}
+	// we can leave logger enabled only for text format
+	if serverConfig.LogFormat != config.LOG_FORMAT_TEXT {
+		// disable logging until we implement zerolog logger for ORM
+		ormConfig.Logger = logger.Default.LogMode(logger.Silent)
+	} else {
+		// output all the SQL queries
+		ormConfig.Logger = logger.Default.LogMode(logger.Info)
```

Comment:
Will it output them into the `INFO` level? I believe `DEBUG` would be more reasonable.

Reply:
I agree, but the ORM's docs don't have the option of DEBUG log level. The lowest I found was INFO. This is acceptable for this situation though. 

The ORM's logger outputs logs in plain-text format, but we use JSON as default production format. So what I've done (for now) is I disabled ORM logs in cases when we use JSON (good for production and is default format), and enabled them with INFO severity (the lowest I found) for when we output logs in plain text (I used the plain-text logs in development, because they are great there :D )

Reply:
I disagree that the INFO level is acceptable for such messages, but let's leave it for now and check how it will go.